modules/phase_field/include/materials/EpsAndMAndLAnisoGauss.h

```diff
@@ -0,0 +1,139 @@
+// 5D Gaussian anisotropy material object
+// Material properties adding anisotropy to epsilon [or also called k], m and L.
+
+#pragma once
+
+#include "ADMaterial.h"
+#include "MooseMesh.h"
+#include "MathUtils.h"
+#include "GrainTrackerInterface.h"
+#include <vector>
+
+class EpsAndMAndLAnisoGauss : public ADMaterial
+{
+public:
+  static InputParameters validParams();
+
+  EpsAndMAndLAnisoGauss(const InputParameters & parameters);
+
+protected:
+  virtual void computeQpProperties() override;
+
+private:
+  void readQuaternionFile();
+  void readLibraryFile();
+
+  // Grains orientation quaternions.
+  std::vector<double> _qwR, _qxR, _qyR, _qzR;
+  // Normalized_Axis_X  Normalized_Axis_Y Normalized_Axis_Z  Angle(Radian) GB_Normal_X  GB_Normal_Y
+  // GB_Normal_Z  Minimun_GB_energy(J/m^2) Minimun_L(nm^3/eV.ns).
+  std::vector<double> _vbalibx, _vbaliby, _vbalibz, _thetabalib, _miubalibx, _miubaliby, _miubalibz,
+      _MinimaEnergy, _MinimaL;
+
+  // Names of material properties.
+  const MaterialPropertyName _eps_name;        // The name of the anisotropic epsilon - AD.
+  const MaterialPropertyName _eps_EN_name;     // The name of the anisotropic epsilon - NoAD.
```

Comment:
Why different AD and non-AD names?

Reply:
I was using TotalFreeEnergy auxkernel that required non-AD materials. But, I made a new AD version ADTotalFreeEnergy; so I do not have anymore non-AD materials.